src/modules/csm/typings.py

```diff
@@ -92,31 +97,44 @@ def dump(
                 perf_data.included += 1 if validator['included'] else 0
                 self.aggr_per_val[key] = perf_data
 
-        with open(self.filename(self.l_slot), mode="wb") as f:
+        with self.buffer.open(mode="wb") as f:
             pickle.dump(self, f)
 
+        os.replace(self.buffer, self.file(self.l_slot, self.r_slot))
+
     @classmethod
-    def try_read(cls, l_slot: SlotNumber) -> Self | None:
+    def try_read(cls, l_slot: SlotNumber, r_slot: SlotNumber) -> Self:
         """Used to restore the object from the persistent storage."""
 
-        filename = cls.filename(l_slot)
-        obj = None
+        file = cls.file(l_slot, r_slot)
+        obj: Self | None = None
 
         try:
-            with open(filename, mode="rb") as f:
+            with file.open(mode="rb") as f:
                 obj = pickle.load(f)
+                assert obj
+
+                logger.info({"msg": f"{repr(obj)} read from {file.absolute()}"})
+
                 # TODO: To think about a better way to check for schema changes.
                 if cls.schema() != obj.__schema__:
                     raise ValueError("Schema mismatch")
         except Exception as e:  # pylint: disable=broad-exception-caught
-            logger.info({"msg": f"Unable to restore FramePerformance instance from {filename}", "error": str(e)})
+            logger.info({"msg": f"Unable to restore FramePerformance instance from {file.absolute()}", "error": str(e)})
 
-        return obj
+        return obj or cls(l_slot=l_slot, r_slot=r_slot)
 
     @property
     def is_coherent(self) -> bool:
         return (self.r_slot - self.l_slot) // 32 == len(self.processed_epochs)
 
-    @staticmethod
-    def filename(l_slot: SlotNumber) -> str:
-        return f"{l_slot}.pkl"
+    @classmethod
+    def file(cls, l_slot: SlotNumber, r_slot: SlotNumber) -> Path:
+        return Path(f"{l_slot}_{r_slot}").with_suffix(cls.EXTENSION)
+
+    @property
+    def buffer(self) -> Path:
+        return self.file(self.l_slot, self.r_slot).with_suffix(".buf")
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}(l_slot={self.l_slot},r_slot={self.r_slot})"
```

Comment:
let’s move them up, closer to  `__post_init__`
quite confusing to read the code above before the implementations of methods and properties